osc.js

```diff
@@ -1,92 +1,122 @@
-const instance_skel = require('../../instance_skel')
-const GetUpgradeScripts = require('./upgrades')
+const { InstanceBase, Regex, runEntrypoint } = require('@companion-module/base')
+const UpgradeScripts = require('./upgrades')
 const OSC = require('osc')
 
-class instance extends instance_skel {
-	constructor(system, id, config) {
-		// super-constructor
-		super(system, id, config)
+class SongbeamerInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
 	}
 
-	GetUpgradeScripts = GetUpgradeScripts
+	/**
+	 * This is a method that initializes the instance for companion
+	 * it needds to set status after successful execution
+	 */
+	async init(config) {
+		this.config = config
+		this.updateStatus('connecting')
+		this.osc_server_init()
+
+		this.log('debug', 'Starting definition of actions feedbacks and variables')
+		this.updateActions()
+		this.updateFeedbacks()
+		this.updateVariables()
+
+		this.updateStatus('ok')
+	}
+
+	/**
+	 * When module gets deleted
+	 */
+	async destroy() {
+		this.log('debug', 'destroy method started')
+		this.osc.close()
+		this.log('debug', 'osc closed')
+		delete this.osc
+		this.log('debug', 'this.osc deleted')
+		this.log('debug', 'destroy')
+	}
 
 	/**
 	 * This is a method is executed on config changes
 	 * It will restart the OSC server if it's configuration changed
 	 */
-	updateConfig(config) {
+	async configUpdated(config) {
 		this.config = config
 		if (this.config.port != this.osc.options.localPort || this.config.host != this.osc.remoteAddress) {
-			this.debug('host or port configuration changed - reloading osc server')
+			this.log('debug', 'host or port configuration changed - reloading osc server')
 			this.osc_server_init()
 		}
 	}
 
 	/**
-	 * This is a method that initializes the instance for companion
-	 * it needds to set status after successful execution
+	 * request updates, subscription expires every 10 seconds
 	 */
-	init() {
-		this.osc_server_init()
-
-		this.actions()
-		this.feedbacks()
-		this.variables()
-		//TODO #1 Initialise variables
-		this.setVariable('presentation_state', 'Not Checked')
-
-		this.status(this.STATE_OK)
+	osc_update_polling() {
+		this.osc.send({
+			address: '/xremote',
+			args: [],
+		})
+		//this.log('debug', 'heartbeat send')
 	}
 
 	/**
 	 * Return config fields for web config
 	 */
-	config_fields() {
+	getConfigFields() {
 		return [
 			{
 				type: 'textinput',
 				id: 'host',
 				label: 'Target IP',
 				width: 8,
 				regex: this.REGEX_IP,
+				useVariables: true,
```

Comment:
`useVariables` should not be used for `host` and `port` config fields.  Config fields using that implementation have to manually parse the variables (which you don't do in `osc_server_init()`) and don't get any triggers when the variable gets updated.  In this case, if a variable was used for the host, even if you parsed it correctly, would not trigger a reconnect when the variable is changed/updated.